src/components/dashboard/modals/approveModal/approveModal.jsx

```diff
@@ -0,0 +1,70 @@
+const React = require('react');
+
+module.exports = class ApproveModal extends React.Component {
+  constructor() {
+    super();
+
+    this.approveResponse = this.approveResponse.bind(this);
+    this.close = this.close.bind(this);
+
+    this.state = {
+      open: false
+    };
+  }
+
+  approveResponse() {
+    const xhr = new XMLHttpRequest();
+
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === 4 && xhr.status === 200) {
+        this.setState({ approved: true });
+        this.close();
+      }
+      else if (xhr.readyState === 4 && xhr.status !== 200) {
+        throw new Error('Approve response failed');
+      }
+    };
+
+    xhr.open('POST', `/responses/${this.props.responseId}/approve`, true);
+    xhr.send();
+  }
+
+  close() {
+    this.setState({ open: false });
+  }
+
+  componentWillUpdate(nextProps) {
+    if (nextProps.responseId !== this.props.responseId) {
+      this.setState({ open: !!nextProps.responseId });
+    }
+  }
+
+  render() {
+    const { responseId } = this.props;
+
+    if (this.state.open) {
```

Comment:
You can return this.state.open && (...jsx) || null
Even better, factor out:
const showModal = this.state.open || null and then return showModal && (...jsx)